flocker/acceptance/test_api.py

```diff
@@ -369,6 +369,61 @@ def can_connect():
         d.addCallback(check_result)
         return d
 
+    def test_create_container_with_environment(self):
+        """
+        Create a container including environment variables on a single-node
+        cluster.
+        """
+        data = {
+            u"name": "my_env_container",
+            u"host": None,
+            u"image": "busybox:latest",
+            u"environment": {u"ACCEPTANCE_ENV_LABEL": 'acceptance test ok'}
+        }
+        waiting_for_cluster = wait_for_cluster(test_case=self, node_count=1)
+
+        def create_container(cluster, data):
+            data[u"host"] = cluster.nodes[0].address
+            return cluster.create_container(data)
+
+        d = waiting_for_cluster.addCallback(create_container, data)
+
+        def check_result((cluster, response)):
+            self.assertEqual(response, data)
+
+        def inspect_container():
+            try:
+                remote_data = run_SSH(
+                    22, 'root', data[u"host"],
+                    [
+                        b"docker",
+                        b"inspect",
+                        b"flocker--{0}".format(data[u"name"])
+                    ], None
+                )
+            except Exception:
+                return False
+            container_data = loads(remote_data)
```

Comment:
Personally, I'd `return container_data` here and perform all the environment variable parsing in the `verify_environment` function below.

As it stands, `loop_until` will treat an empty dictionary as a signal to loop again, so if the container has somehow been started without any environment or if inspect incorrectly returns an empty environment, the test will time out without a meaningful error.